python/lsst/sphgeom/SConscript

```diff
@@ -1,28 +1,33 @@
 from lsst.sconsUtils import scripts
-scripts.BasicSConscript.pybind11([
-    'angle',
-    'angleInterval',
-    'box',
-    'box3d',
-    'chunker',
-    'circle',
-    'convexPolygon',
-    'curve',
-    'ellipse',
-    'htmPixelization',
-    'interval1d',
-    'lonLat',
-    'matrix3d',
-    'mq3cPixelization',
-    'normalizedAngle',
-    'normalizedAngleInterval',
-    'orientation',
-    'pixelization',
-    'q3cPixelization',
-    'rangeSet',
-    'region',
-    'relationship',
-    'unitVector3d',
-    'utils',
-    'vector3d',
-], addUnderscore=False)
+
+scripts.BasicSConscript.pybind11(
+    nameList=['sphgeom'],
+    extraSrc={'sphgeom': [
+        'angle.cc',
+        'angleInterval.cc',
+        'box.cc',
+        'box3d.cc',
+        'chunker.cc',
+        'circle.cc',
+        'convexPolygon.cc',
+        'curve.cc',
+        'ellipse.cc',
+        'htmPixelization.cc',
+        'interval1d.cc',
+        'lonLat.cc',
+        'matrix3d.cc',
+        'mq3cPixelization.cc',
+        'normalizedAngle.cc',
+        'normalizedAngleInterval.cc',
+        'orientation.cc',
+        'pixelization.cc',
+        'q3cPixelization.cc',
+        'rangeSet.cc',
+        'region.cc',
+        'relationship.cc',
+        'unitVector3d.cc',
+        'utils.cc',
+        'vector3d.cc',
+    ]},
+    addUnderscore=False
+)
```

Comment:
I think [the SConscript in `geom`](https://github.com/lsst/geom/blob/master/python/lsst/geom/SConscript) is the recommended way to to write this out; it's subtly different in a way that probably doesn't matter right now but might better avoid to what we might want to deprecate and remove some day.

You'll note that the files there also have a different naming convention.  I'm not sure it's worth the effort to fix them on this ticket, but the geom ones do match the dev guide.  Sadly, the dev guide seems to be flat-out wrong in several ways about what the `SConscript` should look like.